### DISCUSSION TITLE
Contour plot projection not showing properly in matplotlib 3d plotting

### DISCUSSION THREAD

**Discussion by Pushtogithub23** [community/community#114767] [Programming Help]
### Body

I'm trying to plot a contour of a random walker's position in a 3D space on the xy, xz, and yz planes. The plot looks fine on the xy plane, but on the xz and yz planes, the distribution appears strange and is centred around the zero-line. I'm using Jupyter Notebook (Python). Could you suggest a solution to this problem and provide a new plot if it's fixed?

` import numpy as np
import matplotlib.pyplot as plt
from matplotlib import cm

stepsize = 0.5
num_steps = 250
num_trials = 10**3

final_path = []

for _ in range(num_trials+1):
    pos = np.array([0.0,0.0,0.0])
    for _ in range(num_steps+1):
        pos= pos+  np.random.normal(0,0.5,3)
    final_path.append(list(pos))

final_path = np.array(final_path)

x = final_path[:,0]
y = final_path[:,1]
z = final_path[:,2]

fig = plt.figure(figsize = (15,12)) 
ax = fig.add_subplot(projection = '3d', computed_zorder = False)
ax.scatter(x,y,z, c = 'blue', s = 40, zorder = 2)

ax.scatter(x,y, zs = min(z), zdir = 'z', color = 'black',s = 8, zorder = 1)
ax.scatter(x,z, zs = max(y), zdir = 'y', color = 'black',s = 8, zorder = 1)
ax.scatter(y,z, zs = min(x), zdir = 'x', color = 'black',s = 8, zorder = 1)

counts_xy, xedges, yedges = np.histogram2d(x, y, bins=25)
x_grid_xy, y_grid_xy = np.meshgrid(np.linspace(min(x), max(x), 25), np.linspace(min(y), max(y), 25))

counts_xz, xedges, zedges = np.histogram2d(x, z, bins=25)
x_grid_xz, z_grid_xz = np.meshgrid(np.linspace(min(x), max(x), 25), np.linspace(min(z), max(z), 25))

counts_yz, yedges, zedges = np.histogram2d(y, z, bins=25)
y_grid_yz, z_grid_yz = np.meshgrid(np.linspace(min(y), max(y), 25), np.linspace(min(z), max(z), 25))

ax.contourf(x_grid_xy, y_grid_xy, counts_xy.T, zdir='z', offset=min(z), levels = 25, cmap='Reds', zorder=0)
ax.contourf(x_grid_xz, z_grid_xz, counts_xz.T, zdir='y', offset=max(y), levels = 25, cmap='Greens', zorder=0)
ax.contourf(y_grid_yz, z_grid_yz, counts_yz.T, zdir='x', offset=min(x), levels = 25, cmap='Blues', zorder=0)

ax.set_xlim(min(x),max(x))
ax.set_ylim(min(y),max(y))
ax.set_zlim(min(z),max(z))
ax.set_xlabel('X')
ax.set_ylabel('Y')
ax.set_zlabel('Z')
ax.set_title('Walker\'s Position in 3D Space')

plt.show()`

<img width="369" alt="image" src="https://github.com/community/community/assets/147816417/a5072c5a-a109-49ac-a47d-3a50ca137107">

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by mapluisch** [comment]
Hey,

your problem lies in the way you call the `ax.contourf`funcs. When you designate the `zdir` to, eg, `x`, you should also format the input parameters accordingly. Otherwise you'll basically look at the distribution "from the side", which is more akin to a typical histogram I guess :)

So in your case:

``` python
ax.contourf(x_grid_xy, y_grid_xy, counts_xy.T, zdir='z', offset=min(z), levels=25, cmap='Reds', zorder=0)
ax.contourf(x_grid_xz, counts_xz.T, z_grid_xz, zdir='y', offset=max(y), levels=25, cmap='Greens', zorder=0)
ax.contourf(counts_yz.T, y_grid_yz, z_grid_yz, zdir='x', offset=min(x), levels=25, cmap='Blues', zorder=0)
```

Also a tip for future reference, use three backticks (```) for a code block, makes it easier for others to copy & paste your code (and also keeps indentation levels).

**Reply by Pushtogithub23** [reply]
Oh, yes, that's right! Thanks for the help :)

**Reply by mapluisch** [reply]
Sure, you're welcome :)
